src/org/thoughtcrime/securesms/qr/QrActivity.java

```diff
@@ -13,25 +20,39 @@
 import androidx.viewpager.widget.ViewPager;
 
 import com.google.android.material.tabs.TabLayout;
+import com.google.zxing.BinaryBitmap;
+import com.google.zxing.MultiFormatReader;
+import com.google.zxing.NotFoundException;
+import com.google.zxing.Result;
+import com.google.zxing.RGBLuminanceSource;
+import com.google.zxing.common.HybridBinarizer;
 
 import org.thoughtcrime.securesms.BaseActionBarActivity;
 import org.thoughtcrime.securesms.R;
+import org.thoughtcrime.securesms.mms.AttachmentManager;
 import org.thoughtcrime.securesms.permissions.Permissions;
 import org.thoughtcrime.securesms.util.DynamicLanguage;
 import org.thoughtcrime.securesms.util.DynamicNoActionBarTheme;
 import org.thoughtcrime.securesms.util.DynamicTheme;
+import org.thoughtcrime.securesms.util.Util;
 import org.thoughtcrime.securesms.util.ViewUtil;
 
+import java.io.FileNotFoundException;
+import java.io.InputStream;
+
 public class QrActivity extends BaseActionBarActivity {
 
     private final DynamicTheme dynamicTheme = new DynamicNoActionBarTheme();
     private final DynamicLanguage dynamicLanguage = new DynamicLanguage();
 
+    private final static String TAG = QrActivity.class.getSimpleName();
+    private final static int REQUEST_CODE_IMAGE = 9;
```

Comment:
as this is an app-wide-magic number, maybe use something easier grep'able with more digits (31331 ... sth. that does not exist in code currently :)

however, it was doing for the other selectImage() calls only partly as well, but still.

Reply:
I will use 46243, that means `i-m-a-g-e` in a phone dial pad / keyboard :D